tools/graph_loading/construct_graph.py

```diff
@@ -0,0 +1,610 @@
+""" Preprocess the input data """
+from multiprocessing import Process
+import multiprocessing
+import glob
+import os
+import dgl
+import json
+import pyarrow.parquet as pq
+import pyarrow as pa
+import numpy as np
+from functools import partial
+import argparse
+
+from transformers import BertTokenizer
+import torch as th
+
+##################### The I/O functions ####################
+
+def read_data_parquet(data_file):
+    """ Read data from the parquet file.
+
+    A row of a multi-dimension data is stored as an object in Parquet.
+    We need to stack them to form a tensor.
+
+    Parameters
+    ----------
+    data_file : str
+        The parquet file that contains the data
+
+    Returns
+    -------
+    dict : map from data name to data.
+    """
+    table = pq.read_table(data_file)
+    pd = table.to_pandas()
+    data = {}
+    for key in pd:
+        d = np.array(pd[key])
+        # For multi-dimension arrays, we split them by rows and
+        # save them as objects in parquet. We need to merge them
+        # together and store them in a tensor.
+        if d.dtype.hasobject:
+            d = [d[i] for i in range(len(d))]
+            d = np.stack(d)
+        data[key] = d
+    return data
+
+def write_data_parquet(data, data_file):
+    """ Write data in parquet files.
+
+    Normally, Parquet cannot support multi-dimension arrays.
```

Comment:
I don't understand your comment or which part is confusing.
Parquet format doesn't have native support multi-dimension arrays. It can only support vectors. It has nothing to do with the data type of each individual element. I'm not sure if this is what you mean in the comment.